src/Data/toClientParams.ts

```diff
@@ -0,0 +1,41 @@
+import { provideDataClass } from 'scrivito'
+
+type DataConnection = Parameters<typeof provideDataClass>[1]['connection']
+type IndexCallback = NonNullable<DataConnection['index']>
+type IndexParams = Parameters<IndexCallback>[0]
+
+interface OperatorSpec {
+  operator: 'equals' | 'notEquals'
+  opCode: 'eq' | 'neq'
+  value: string
+}
+
+export function toClientParams(params: IndexParams) {
```

Comment:
* [ ] Missing a return type.

Reply:
Why is this needed? The derived types look fine to me.